File: WebClientIS/WebClientIS/Models/Paciente.cs
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Data.SqlClient;
using System.Linq;
using System.Net.Http;
using System.Net.Http.Headers;
using System.Text;
using System.Threading.Tasks;
using System.Web.Mvc;

namespace APIIS.Models
{
    public class Paciente
    {
        public int IDU { get; set; }

        public string CodigoPac { get; set; }

        public int Idade { get; set; }

        public string Sexo { get; set; }

        public string Estado { get; set; }

        public bool Risco { get; set; } 

        public string Morada { get; set; }

        public string CodigoPostal { get; set; }


        public Paciente()
        {
            IDU = 0;
            CodigoPac = string.Empty;
            Idade = 0;
            Sexo = string.Empty;
            Estado = "Não infetado";
            Risco = false;
            Morada = string.Empty;
            CodigoPostal = string.Empty;

        }


        public List<Paciente> GetPacientes()
        {
            List<Paciente> lista = new List<Paciente>();

            HttpClient client = new HttpClient();

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/Paciente/" );

            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));

            HttpResponseMessage response = client.GetAsync("").Result;

         
                var products = response.Content.ReadAsStringAsync().Result;

                var array = JArray.Parse(products);

                foreach (var obj in array)
                {
                    
                        lista.Add(obj.ToObject<Paciente>());
                    
                 
                }

            

            return lista;

        }
        public IEnumerable<SelectListItem> TypeList
        { 
            get
            {
                List<SelectListItem> x = new List<SelectListItem>();
                  Conselho c = new Conselho();
                List<Conselho> lc = new List<Conselho>();
                lc = c.GetConselhos();
                foreach(Conselho conselho in lc)
                {
                    x.Add(new SelectListItem { Text = conselho.Nome, Value = conselho.Nome });
                    
                }
                return x;
            }
        }
        public bool CreatePaciente()
        {
            HttpClient client = new HttpClient();

            var values = new Dictionary<string, string>
            {
                { "IDU", IDU.ToString() },
                { "CodigoPac", CodigoPac },
                { "Idade", Idade.ToString() },
                { "Sexo", Sexo },
                { "Estado", Estado },
                { "Risco", Risco.ToString() },
                { "Morada", Morada },
                { "CodigoPostal", CodigoPostal },
            };

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/Paciente");

            var jsonString = JsonConvert.SerializeObject(values);
            var content = new StringContent(jsonString, Encoding.UTF8, "application/json");

            var resposta = client.PostAsync(client.BaseAddress.ToString(), content).Result;

            if (resposta.IsSuccessStatusCode)
            {
                return true;
            }
            else
            {
                return false;
            }
        }

        public bool UpdatePacienteEstado(int idu, object value)
        {
            HttpClient client = new HttpClient();

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/Paciente/" + idu.ToString());

            var jsonString = JsonConvert.SerializeObject(value.ToString());
            var content = new StringContent(jsonString, Encoding.UTF8, "application/json");

            var resposta = client.PutAsync(client.BaseAddress.ToString(), content).Result;

            if (resposta.IsSuccessStatusCode)
            {
                return true;
            }
            else
            {
                return false;
            }
        }

        public bool UpdatePacienteMorada(int idu, object value)
        {
            HttpClient client = new HttpClient();

            client.BaseAddress = new Uri("http://6f85a52b4f05.ngrok.io/api/Paciente/" + idu.ToString() + "/Morada");

            var jsonString = JsonConvert.SerializeObject(value.ToString());
            var content = new StringContent(jsonString, Encoding.UTF8, "application/json");

            var resposta = client.PutAsync(client.BaseAddress.ToString(), content).Result;

            if (resposta.IsSuccessStatusCode)
            {
                return true;
            }
            else
            {
                return false;
            }
        }


    }
}

File: MobileClient/sampleproject/lib/Models/Paciente.dart
import 'dart:typed_data';
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;
import 'dart:convert' show json;
import 'package:scoped_model/scoped_model.dart';


class Paciente extends Model{

  int idu;
  String codigopac;
  int idade;
  String sexo;
  String estado;
  bool risco;
  String morada;
  String codigopostal;

  Paciente({int iduu,String cp, int age,String sex,String state,bool risk, String mor,String codigop})
  {
    this.idu = iduu;
    this.codigopac = cp;
    this.idade = age;
    this.sexo = sex;
    this.estado = state;
    this.risco = risk;
    this.morada = mor;
    this.codigopostal = codigop;

  }

  factory Paciente.fromJson(Map<String,dynamic> json){

    return Paciente(
      iduu: json['idu'] as int,
      cp: json['codigoPac'] as String,
      age: json['idade'] as int,
      sex:json['sexo'] as String,
      state: json['estado'] as String,
      risk: json['risco'] as bool,
      mor: json['morada'] as String,
      codigop:json['codigoPostal'] as String
          
    );

  }

  Future<List> getpacientes() async {
  
  http.Response response = await http.get(
    Uri.encodeFull("http://6f85a52b4f05.ngrok.io/api/Paciente"),
    headers:{
      "Accept":"application/json"
    }
  );

  List lista = json.decode(response.body);

  List<Paciente> listau = new List<Paciente>();

  
  for(int i = 0; i < lista.length; i++)
    listau.add(Paciente.fromJson(lista[i]));
  

  return listau;
}

Future<List<Paciente>> getpaciente(int idu,String what, String value) async {
  
  http.Response response = await http.get(
    Uri.encodeFull("http://6f85a52b4f05.ngrok.io/api/Paciente/"+idu.toString()+"/"+what+"/"+value),
    headers:{
      "Accept":"application/json"
    }
  );

  List lista = json.decode(response.body);

  List<Paciente> listau = new List<Paciente>();

  
  for(int i = 0; i < lista.length; i++)
    listau.add(Paciente.fromJson(lista[i]));
  
  return listau;
}

Future<int> createpaciente(Paciente pac) async {

  var url ='http://6f85a52b4f05.ngrok.io/api/Paciente';
  var body = json.encode(<String, String>{
      'IDU': pac.idu.toString(),
      'CodigoPac':pac.codigopac,
      'Idade':pac.idade.toString(),
      'Sexo':pac.sexo,
      'Estado':pac.estado,
      'Risco':pac.risco.toString(),
      'Morada':pac.morada,
      'CodigoPostal':pac.codigopostal

    });


  http.Response response = await http.post(url,
      headers: {"Content-Type": "application/json"},
      body: body
  );

  return response.statusCode;
}

Future<int> updatePaciente(int idu,String value) async {

  var url ='http://6f85a52b4f05.ngrok.io/api/Paciente/' + idu.toString();
  var body = json.encode(value);


  http.Response response = await http.put(url,
      headers: {"Content-Type": "application/json"},
      body: body
  );

  return response.statusCode;
}


}